### ISSUE TITLE
Training on very small dataset (~10 images)

### ISSUE THREAD

**Issue by ECer23** [AlexeyAB/darknet#2881]
Is there any tip on training very small dataset? 

I want to detect one specific object instance (not a category) with <20 training images. I change config file as instructed in README, but the loss drop very fast and quickly overfitting.

**Comment by nbkhuong** [comment]
This should not happen so quickly right, not after 2k iterations I suppose, since 10k images set is not a small set for training. You might also be interested in looking at mAP on your validation set to see if it increases over the training. 

I am also interested in detecting small object and got some problems as well, can you share how small is your object with your anchors_calc (see the instructions on the git page)

**Comment by ECer23** [comment]
@nbkhuong thanks for your advice! But the "small dataset" I mean is the number of training images is small, not the object is small. I have only 10 images for training.

I have recalculate the anchor, increase the resolution, and fine tune on coco pre-trained weights. But the loss drop to < 1 in around 100 iterations. Any advice on how to prevent overfitting on small datasets(few-shot training)?

**Comment by nbkhuong** [comment]
Have you tried to test your nets on some tests? It's probably due to the data you collect for training, the images contain too many features that have approximately similar properties (same sizes, captured from same view, etc...). calc_anchors will help you better with the visualization of the set.

**Comment by ECer23** [comment]
Yes I've tried but failed.

**Comment by AlexeyAB** [comment]
I think the random blurring of background can help for small datasets. I will try to add it to the data augmentation later.

What objects do you use, which you can't collect more than 10 images for training?

**Comment by ECer23** [comment]
I use objects in [YCB dataset](http://www.ycbbenchmarks.com). Photos are captured by myself. The objective is to detect a specific object (like tomato soup can) with as few training images as possible.

Will it be helpful if I segment the objects out and paste it to random background?

**Comment by AlexeyAB** [comment]
@ECer23 

> Will it be helpful if I segment the objects out and paste it to random background?

Yes.
You can generate different images with different backgrounds/scales/angles/slopes/illuminations/... of objects.

Do you use Texture-mapped 3D mesh models from YCB dataset?

**Comment by ECer23** [comment]
@AlexeyAB 

> Do you use Texture-mapped 3D mesh models from YCB dataset?

No, I take photos from real object.
